multimodal/src/autogluon/multimodal/optimization/utils.py

```diff
@@ -799,3 +799,34 @@ def compute_probability(
         prob = 1 - prob
 
     return prob
+
+
+# Reference: https://github.com/lenscloth/RKD/blob/0a6c3c0c190722d428322bf71703c0ae86c25242/metric/utils.py#L6
+def pdist(embeddings: Optional[torch.Tensor], squared: Optional[bool] = False, eps: Optional[float] = 1e-12):
+    """
+    Compute pairwise Euclidean distances between embeddings in n-dimensional space.
+
+    Parameters
+    ----------
+    embeddings
+        The embeddings to compute pairwise distance between.
+    squared
+        If the result is square of Euclidean distance.
+    eps
+        Min value of each entry.
+
+    Returns
+    -------
+    Pairwise Euclidean distances.
+    """
+    embeddings_square = embeddings.pow(2).sum(dim=1)
+    prod = embeddings @ embeddings.t()
+    res = (embeddings_square.unsqueeze(1) + embeddings_square.unsqueeze(0) - 2 * prod).clamp(min=eps)
+
+    if not squared:
+        res = res.sqrt()
+
+    res = res.clone()
```

Comment:
Just curious, why do we need to clone?

Reply:
This seemed to be redundant but I was not sure. I'll checked several repos and all of them have this line, but I guess they copied from the same source. I'll remove this and then test run.

Reply:
There is a Runtime error if this is removed:

```
RuntimeError: one of the variables needed for gradient computation has been modified by an inplace operation: [torch.cuda.FloatTensor [8, 8]], which is output 0 of SqrtBackward0, is at version 1; expected vers
ion 0 instead. Hint: enable anomaly detection to find the operation that failed to compute its gradient, with torch.autograd.set_detect_anomaly(True).
```

Reply:
I think we may also fix by `res[range(len(embeddings)), range(len(embeddings))] *= 0` in line 830.